src/test/java/org/entur/lamassu/model/id/predicate/IdPredicateBuilderTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *
+ *
+ *  * Licensed under the EUPL, Version 1.2 or – as soon they will be approved by
+ *  * the European Commission - subsequent versions of the EUPL (the "Licence");
+ *  * You may not use this work except in compliance with the Licence.
+ *  * You may obtain a copy of the Licence at:
+ *  *
+ *  *   https://joinup.ec.europa.eu/software/page/eupl
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the Licence is distributed on an "AS IS" basis,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the Licence for the specific language governing permissions and
+ *  * limitations under the Licence.
+ *
+ */
+
+package org.entur.lamassu.model.id.predicate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class IdPredicateBuilderTest {
+
+  @Test
+  void testCodespace() {
+    IdPredicate predicate = IdPredicateBuilder.newInstance().withCodespace("AAA").build();
+
+    assertTrue(predicate.test("AAA:Network:123"));
+    assertFalse(predicate.test("BBB:Network:123"));
+    assertTrue(predicate.test("AAA:xyz:123"));
+    assertFalse(predicate.test("CCC:z:123"));
+
+    assertFalse(predicate.test("AAAANetwork:123"));
+
+    // non-validating
+    assertTrue(predicate.test("AAA:X"));
+
+    assertFalse(predicate.test("AAB:Network:123"));
+    assertFalse(predicate.test("ABA:Network:123"));
+    assertFalse(predicate.test("BAA:Network:123"));
+  }
+
+  @Test
+  void testType() {
```

Comment:
Could you add some special case tests for colons as well, as these might be the most difficult to get right

Reply:
Values can contain colon, I added a test for it here: https://github.com/entur/lamassu/pull/516/commits/84ae36bf26fa5c5cecfa25871bb0f3c099e2afac

I think there might be some confusion. The IdPredicate exists to be able to verify that some ID conforms to the codespace/type pattern as described by the values you give to the builder.

So given the following IdPredicate produced by a builder:

```java
    IdPredicate predicate = IdPredicateBuilder
      .newInstance()
      .withCodespace("AAA")
      .withType("Network")
      .build();
```

you can test that some string conforms to the pattern "AAA:Network:<some_value>". It does not however, validate that the *value* itself is valid.